src/module/io.wren

```diff
@@ -107,6 +108,44 @@ foreign class File {
     return Scheduler.await_ { realPath_(path, Fiber.current) }
   }
 
+  static dirname(path) {
+    var separator = Platform.directorySeparator
```

Comment:
I find this problematic.  `dirname` on Linux or Windows should be able to handle paths from either platform.  I don't think it's clear at all that the ONLY paths these functions will ever be called to handle are those on the platform they are running on.

There are pure string processing functions, their behavior should not be conditional based on the OS.

Reply:
If this assumption is already baked into the existing File/Dir functions it may be ok "for now" though.  But when I rewrote the CLI in Wren I removed all such assumptions - the resolver Wren kernel (and it's Path class) doesn't have access to `Platform.directorySeparator`... so if it's given a path in the unix form, it works that way, if it's given a path in the Windows form, it works that way.

For resolution of REAL actual paths it calls the UVlibs `realpath` implementation - which we do use to resolve actual files on the disk...

Reply:
This also simplifies testing since it removes the conditionals...

Reply:
Perhaps this works for now.